.circleci/config.yml

```diff
@@ -56,72 +18,117 @@ executors:
       - image: cimg/node:22.14.0
     working_directory: *working_directory
 
+commands:
+  pnpm_install:
+    description: Installs dependencies using PNPM
+    parameters:
+      pnpm_store_dir:
+        type: string
+        default: /home/circleci/.local/share/pnpm/store
+      pnpm_install_directory:
+        type: string
+        default: /home/circleci/bin
+      command:
+        type: string
+        default: pnpm install --frozen-lockfile
+    steps:
+      - restore_cache:
+          name: Restoring pnpm cache
+          keys:
+            # Delete and re-create a increased `CACHE_VERSION_PNPM` variable in CircleCI's project configuration.
+            # https://app.circleci.com/settings/project/github/commercetools/nodejs/environment-variables
+            - pnpm-cache-{{ .Environment.CACHE_VERSION_PNPM }}-{{ checksum "pnpm-lock.yaml" }}
+      - run:
+          name: Ensuring pnpm installation directory exists
+          command: mkdir -p << parameters.pnpm_install_directory >>
+      - run:
+          name: Exporting << parameters.pnpm_install_directory >> to PATH
+          command: echo 'export PATH=$PATH:<< parameters.pnpm_install_directory >>' >> $BASH_ENV
+      - run:
+          name: Installing pnpm via corepack
+          command: |
+            corepack enable --install-directory << parameters.pnpm_install_directory >>
+            corepack prepare --activate
+      - run:
+          name: Setting pnpm store
+          command: pnpm config set store-dir << parameters.pnpm_store_dir >>
+      - save_cache:
+          name: Saving pnpm cache
+          key: pnpm-cache-{{ .Environment.CACHE_VERSION_PNPM }}-{{ checksum "pnpm-lock.yaml" }}
+          paths:
+            - node_modules
+      - run:
+          name: Installing project dependencies
+          command: << parameters.command >>
+
 jobs:
   install_test_node_18:
     executor: node_18
     steps:
       - checkout
-      - restore_cache: *restore_yarn_cache_node_18
-      - run: *yarn_install
-      - save_cache: *save_yarn_cache_node_18
+      - pnpm_install
+      - run:
```

Comment:
I suggest not to build as a post install. It's not needed and frustrating. If we don't do that we must do it here.